app/root.tsx

```diff
@@ -44,3 +63,53 @@ export default function App() {
     </html>
   );
 }
+
+export function ErrorBoundary() {
+  const error = useRouteError();
+
+  if (isRouteErrorResponse(error)) {
+    if (error.status == 404) {
+      return (
+        <Shell title="An Error Occurred">
+          <ErrorMessage>
+            <div className="flex">
+              <img
+                className="h-0 w-0 collapse sm:visible sm:h-60 sm:w-60"
+                src="/imgs/404-not-found.png"
+                alt="Page not found"
+              />
+              <h1>The page you are looking for does not exist.</h1>
+            </div>
+          </ErrorMessage>
+        </Shell>
+      );
+    }
+    return (
+      <Shell title="An Error Occurred">
+        <ErrorMessage>
+          <h1>
+            {error.status} {error.statusText}
+          </h1>
+          <p>{error.data}</p>
+        </ErrorMessage>
+      </Shell>
+    );
+  } else if (error instanceof Error) {
+    return (
+      <Shell title="Something went wrong">
+        <ErrorMessage>
+          <h1>Error</h1>
+          <p>{error.message}</p>
```

Comment:
Just food for thought - For security reasons, we may not want to default to showing a raw error message when we go to production. For example, an error could be thrown by our db that includes info about which ORM and database we're using, which in turn gives a potential hacker more info to attack our web app.

Reply:
It looks like Remix already sanitizes errors for us so we don't need to worry about this. Here is a reference link - https://remix.run/docs/en/main/guides/errors#error-sanitization

Reply:
Great catch 👍